### PR TITLE
Remove old API from docs

### PR DESCRIPTION
**Pull request by oleurud**
Solves https://github.com/CartoDB/cartoframes/issues/873 & https://github.com/CartoDB/cartoframes/issues/790

Docs ---> https://cartoframes.readthedocs.io/en/documentation-remove-old-api-references

- [x] Readme
- [x] Old classes references
- [x] Credentials / authorization
- [x] Examples
- [x] Cookbooks
- [x] ETL

I have removed the `How to get a matplotlib figure with four maps` because I think it is not possible right now https://github.com/CartoDB/cartoframes/issues/877

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -272,24 +272,26 @@ CARTO Credential Management
...
 
-The most common way to input credentials into cartoframes is through the `Context`, as below. Replace `{your_user_name}` with your CARTO username and `{your_api_key}` with your API key, which you can find at ``https://{your_user_name}.carto.com/your_apps``.
+The most common way to input credentials into cartoframes is through the `set_default_credentials` method, as below. Replace `{your_user_name}` with your CARTO username and `{your_api_key}` with your API key, which you can find at ``https://{your_user_name}.carto.com/your_apps``.
 
 .. code:: python
 
-    from cartoframes.auth import Context
-    con = Context(
-        base_url='https://{your_user_name}.carto.com',
+    from cartoframes.auth import set_default_credentials
```
Comment (elenatorro) [inline README.rst]: Should we mention here that the `api_key` is only required for private datasets? It makes possible to set the credentials by using only the username: `set_default_credentials('{your_user_name}')`. Just a suggestion, don't want to add too much noise in this section.

File: docs/credentials.rst [hunk 2]
```diff
@@ -1,5 +1,35 @@
...
+The most common way to input credentials into cartoframes is through the `set_default_credentials` method, as below. Replace `{your_user_name}` with your CARTO username and `{your_api_key}` with your API key, which you can find at ``https://{your_user_name}.carto.com/your_apps``.
+
+.. code:: python
+
+    from cartoframes.auth import set_default_credentials
+
+    set_default_credentials(
+        username='{your_user_name}',
+        api_key='{your_api_key}'
+    )
```
Comment (elenatorro) [inline docs/credentials.rst]: Same here

File: README.rst (hunk 1 above)
Comment (oleurud) [reply README.rst]: :+1: Done!

File: docs/credentials.rst (hunk 2 above)
Comment (oleurud) [reply docs/credentials.rst]: :+1:

**Pull request merged**